task/04-arrays-tasks.js

```diff
@@ -505,10 +502,15 @@ function distinct(arr) {
  *   }
  */
 function group(array, keySelector, valueSelector) {
-  throw new Error('Not implemented');
+  return array.reduce((map, value) => {
+    let cities = map.get(keySelector(value));
+    if (cities) cities.push(valueSelector(value));
```

Comment:
you must use curly brackets when working with `if else`

Reply:
discussed